packages/flutter/lib/src/services/text_input.dart

```diff
@@ -662,12 +662,19 @@ class TextInputConnection {
   void close() {
     if (attached) {
       SystemChannels.textInput.invokeMethod<void>('TextInput.clearClient');
-      _clientHandler
-        .._currentConnection = null
-        .._scheduleHide();
+      _onConnectionClosed();
+      _clientHandler._scheduleHide();
```

Comment:
add a comment here? i.e. we think it's safe to assume that the platform already hide what needed to hiding if the platform calls us to tell us that we disconnected?

Reply:
I'm not sure I understand...the line you commented is the one that actually does hide the keyboard. Did you mean that we should comment from the other place where `_onConnectionClosed()` is invoked? or something else?